src/validation_subscriber.js

```diff
@@ -0,0 +1,127 @@
+import {CronJob} from 'cron'
+import request from 'superagent'
+import WebSocket from 'ws'
+
+const PEER_PORT_REGEX = /51235/g
+const WS_PORT = '51233'
+
+export default class ValidationSubscriber {
+  constructor() {
+    if (!process.env.PEER_API_URL) {
+      throw new Error('PEER_API_URL required')
+    } else if (!process.env.VALIDATOR_REGISTRY_API_URL) {
+      throw new Error('VALIDATOR_REGISTRY_API_URL required')
+    }
+    this.peerApiUrl = process.env.PEER_API_URL
+    this.validatorRegistryApiUrl = process.env.VALIDATOR_REGISTRY_API_URL
+    this.authUsername = process.env.VALIDATOR_REGISTRY_API_AUTH_USER || ''
+    this.authPassword = process.env.VALIDATOR_REGISTRY_API_AUTH_PASS || ''
+    this.connections = {}
+  }
+
+  async getRippleds() {
+    const response = await request.get(this.peerApiUrl)
+    return response.body
+  }
+
+  submitValidation(validation, reporterPublicKey) {
```

Comment:
Good idea. I wasn't thinking of it since this is only called at the end of the on 'message' websocket event handler.